cmd/dockerd/daemon.go

```diff
@@ -484,7 +484,7 @@ func initRouter(s *apiserver.Server, d *daemon.Daemon, c *cluster.Cluster) {
 		image.NewRouter(d, decoder),
 		systemrouter.NewRouter(d, c),
 		volume.NewRouter(d),
-		build.NewRouter(dockerfile.NewBuildManager(d)),
+		build.NewRouter(manager.NewBuildManager(d, d), d),
```

Comment:
I think these examples are a mistake that we should not repeat. 

The router is supposed to be a thin layer that just translates an HTTP request into a backend request, and then a backend response into an HTTP response. Any more logic than that violates the design.

>> "squash" definitely doesn't belong in the "core builder"

> Can you explain why you think that

I guess definitely was too strong of a word. I think for some definitions of "builder" it is outside the scope. It's a post-processing step that you can perform afterwards. Since it is easy to separate from the rest of the dispatching code, I think it's a good idea to keep it separate.